pkg/client/testclient/fixture.go

```diff
@@ -216,3 +218,46 @@ func (o objects) Add(obj runtime.Object) error {
 
 	return nil
 }
+
+func (o objects) Delete(obj runtime.Object) error {
```

Comment:
Also, if you want to return an error for a particular type, you can set that on api.Status:

   o.Add(&api.Status{Details: &api.StatusDetails{Kind: "pod", ID: "name"}})

That way you can simulate errors on successive calls.  I *think* there's a test case that covers it.

----- Original Message -----
> > @@ -216,3 +218,46 @@ func (o objects) Add(obj runtime.Object) error {
> >  
> >  	return nil
> >  }
> > +
> > +func (o objects) Delete(obj runtime.Object) error {
> 
> I just tested client.Pods(ns).Get("any") and the first return was podA, call
> it again and the second return was podB.  I understand now what you mean by
> sequence.
> 
> also, i will refactor my unit test to use mocks and narrow interfaces more
> like #6871.  Don't need this PR.
> 
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/GoogleCloudPlatform/kubernetes/pull/6928/files#r28551967